src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,52 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> previousNode;
+
+        public Node(Node previousNode, T value) {
```

Comment:
Access modifier. You don't need public access for this class

Reply:
agree, done.